kube/src/lib.rs

```diff
@@ -312,16 +316,32 @@ mod test {
             image: String,
         }
         type PodSimple = Object<PodSpecSimple, NotUsed>;
+
         // use known type information from pod (can also use discovery for this)
         let ar = ApiResource::erase::<Pod>(&());
 
         let client = Client::try_default().await?;
         let api: Api<PodSimple> = Api::default_namespaced_with(client, &ar);
-        for p in api.list(&Default::default()).await? {
-            // run loop to cover ObjectList::iter
-            println!("found pod {} with containers: {:?}", p.name(), p.spec.containers);
+        let mut list = api.list(&Default::default()).await?;
+        // check we can mutably iterate over it
+        for pod in list.iter_mut() {
```

Comment:
Isn't this just regular `Vec` behaviour?

Reply:
the list is actually an `ObjectList` which contains an `items` vector, and the `Iterator` just defers to it in the iterator impl. (it's kind of unnecessary to test due to type guarantees so combined it with another mut thing with the ResourceExt trait to check the whole pipeline worked).

Reply:
Oh right, true.